buildpack.md

```diff
@@ -476,35 +476,62 @@ The purpose of launch is to modify the running app environment using app-provide
    1. **IF** the `CNB_PROCESS_TYPE` environment variable is set,
       1. **IF** the value of `CNB_PROCESS_TYPE` corresponds to a process in `<layers>/launch.toml`, \
-         **THEN** the lifecycle MUST execute the corresponding command in the container using the Bash shell process used to source the `profile.d` scripts.
+         **THEN** the lifecycle MUST choose the corresponding process as the start command.
 
       2. **IF** the value of `CNB_PROCESS_TYPE` does not correspond to a process in `<layers>/launch.toml`, \
          **THEN** launch fails.
 
    2. **IF** the `CNB_PROCESS_TYPE` environment variable is not set,
       1. **IF** there is a process with a `web` process type in `<layers>/launch.toml`, \
-         **THEN** the lifecycle MUST execute the corresponding command in the container using the Bash shell process used to source the `profile.d` scripts.
+         **THEN** the lifecycle MUST choose the corresponding process as the start command.
 
       2. **IF** there is not a process with a `web` process type in `<layers>/launch.toml`, \
          **THEN** launch fails.
+         
+To choose an execution strategy,
+
+1. **IF** the value of `CMD` is chosen as the start command,
+   1. **IF** the length of the value of `CMD` is one,
+      **THEN** the lifecycle MUST invoke the value as a command using Bash
+
+   2. **IF** the length of the value of `CMD` is greater than one,
+      **THEN** the lifecycle MUST invoke the first entry using the `execve` syscall with subsequent entries as arguments.
+
+
+2. **IF** a buildpack-provided process type is chosen as the start command,
+   1. **IF** the process type does not have an `args` field,
+      **THEN** the lifecycle MUST invoke the value of `command` as a command using Bash
+
+   2. **IF** the process type does have an `args` field,
+      **THEN** the lifecycle MUST invoke the first entry using the `execve` syscall with subsequent entries as arguments.
+
+Given the start command and execution strategy,
+
+1. The lifecycle MUST set all buildpack-provided launch environment variables as described in the [Environment](#environment) section.
 
-When executing a process with Bash, the lifecycle SHOULD replace the Bash process in memory with the resulting command process if possible.
+2. If using an execution strategy involving Bash, the lifecycle MUST use a single Bash process to
+   1. source each file in each `<layers>/<layer>/profile.d` directory,
+      1. Firstly, in order of `/bin/build` execution used to construct the OCI image.
+      2. Secondly, in alphabetically ascending order by layer directory name.
+      3. Thirdly, in alphabetically ascending order by file name.
+   2. source `<app>/.profile` if it is present.
+
+3. The lifecycle MUST invoke the start command with the decided execution strategy.
+
+When executing a process with, the lifecycle SHOULD additionally replace the lifecycle process in memory with the command process.
```

Comment:
with `execve`?

Reply:
Yeah, typo. This line is just intended to mean “don’t fork before exec.” Will clarify.

Reply:
Fixed.